### PR TITLE
Add TLS support for LDAP

### PR DESCRIPTION
**Pull request by derek63**

### CODE REVIEW

File: auth/src/main/java/com/redhat/lightblue/rest/auth/ldap/LightblueLdapRoleProvider.java [hunk 1]
```diff
@@ -43,10 +40,10 @@
     InitialLdapContextProvider ldapContextProvider;
     String ldapSearchBase;
 
-    public LightblueLdapRoleProvider(String server, String searchBase, String bindDn, String bindDNPwd) throws NamingException {
+    public LightblueLdapRoleProvider(String server, String searchBase, String bindDn, String bindDNPwd, Boolean useSSL, String trustStore, String trustStorePassword) throws NamingException {
```
Comment (dcrissman) [inline auth/src/main/java/com/redhat/lightblue/rest/auth/ldap/LightblueLdapRoleProvider.java]: Would there be value in keeping both constructors? You wouldn't need to touch LightblueLdapRoleProviderTest then.

File: auth/src/main/java/com/redhat/lightblue/rest/auth/ldap/InitialLdapContextProvider.java [hunk 2]
```diff
@@ -21,7 +20,7 @@
 
     private Hashtable<String, Object> env = new Hashtable<>();
 
-    public InitialLdapContextProvider(String server, String bindDn, String bindDNPwd) {
+    public InitialLdapContextProvider(String server, String bindDn, String bindDNPwd, Boolean useSSL, String trustStore, String trustStorePassword) {
```
Comment (dcrissman) [inline auth/src/main/java/com/redhat/lightblue/rest/auth/ldap/InitialLdapContextProvider.java]: Would there be value in keeping both constructors?

File: auth/src/main/java/com/redhat/lightblue/rest/auth/ldap/InitialLdapContextProvider.java [hunk 3]
```diff
@@ -33,6 +32,11 @@ public InitialLdapContextProvider(String server, String bindDn, String bindDNPwd
         }
         env.put(Context.INITIAL_CONTEXT_FACTORY, "com.sun.jndi.ldap.LdapCtxFactory");
         env.put(Context.PROVIDER_URL, server);
+        if(useSSL) {
+            System.setProperty("javax.net.ssl.trustStore", trustStore);
+            System.setProperty("javax.net.ssl.trustStorePassword", trustStorePassword);
```
Comment (dcrissman) [inline auth/src/main/java/com/redhat/lightblue/rest/auth/ldap/InitialLdapContextProvider.java]: Where/When do these system properties get read back in?

Comment (derek63) [inline auth/src/main/java/com/redhat/lightblue/rest/auth/ldap/InitialLdapContextProvider.java]: The built-in Java LDAP support uses these when you connect to an ldaps:// endpoint.  Using System.setProperty is totally tacky, which is why I opened https://github.com/lightblue-platform/lightblue-rest/issues/257.  However, we don't have time for all that at the moment...

File: auth/src/main/java/com/redhat/lightblue/rest/auth/ldap/LightblueLdapRoleProvider.java (hunk 1 above)
Comment (derek63) [inline auth/src/main/java/com/redhat/lightblue/rest/auth/ldap/LightblueLdapRoleProvider.java]: Perhaps, let me add the default one back, and revert the LightblueLdapRoleProviderTest change, and we can revisit this when we get to changing out the LDAP library (see https://github.com/lightblue-platform/lightblue-rest/issues/257)

**Pull request merged**